porch/repository/pkg/git/package.go

```diff
@@ -139,7 +141,24 @@ func (p *gitPackageRevision) GetUpstreamLock() (kptfile.Upstream, kptfile.Upstre
 				Repo:      repo,
 				Directory: p.path,
 				Ref:       p.revision,
-				Commit:    p.sha.String(),
+				Commit:    p.commit.String(),
 			},
 		}, nil
 }
+
+func (p *gitPackageRevision) getPackageRevisionType() v1alpha1.PackageRevisionType {
+	switch {
+	case p.ref == nil || p.ref.Name().IsTag():
+		return v1alpha1.PackageRevisionTypeFinal
```

Comment:
Does `ref == nil`  mean Final?

Reply:
at this point the field should never be nil. I didn't want to panic the server ... that seemed cruel :)

I'll just turn this into an (internal) error since that's what it would be if it were to occur.